glean-core/rlb/src/net/mod.rs

```diff
@@ -41,11 +41,16 @@ pub(crate) struct UploadManager {
     inner: Arc<Inner>,
 }
 
+const STATE_STOPPED: u8 = 0;
+const STATE_RUNNING: u8 = 1;
+const STATE_SHUTTING_DOWN: u8 = 2;
+
 #[derive(Debug)]
 struct Inner {
     server_endpoint: String,
     uploader: Box<dyn PingUploader + 'static>,
-    thread_running: AtomicBool,
+    thread_running: AtomicU8,
```

Comment:
Yeah, I'm definitely gonna want to make this look more `AtomicEnum`-ish before landing.